### ISSUE TITLE
[2015.05] Cannot look up attributes in a type object

### ISSUE THREAD

**Issue by atweiden** [MoarVM/MoarVM#218]
This occurs in Rakudo 2015.05 release but not in 2015.04. Basic case:

```perl6
use v6;

subset Price of Rat where * >= 0;

my Price %dates_and_prices{Date};
my Price %dates_and_prices_from_file{Date};

%dates_and_prices =
    Date.new("2014-01-01") => 13.45,
    Date.new("2014-01-02") => 12.34,
    Date.new("2014-01-03") => 11.11,
    Date.new("2014-01-04") => 10.09,
    Date.new("2014-01-05") => 9.08,
    Date.new("2014-01-06") => 8.07,
    Date.new("2014-01-07") => 7.06;

my Price %xe{Date} = (%dates_and_prices_from_file, %dates_and_prices);

say %xe.perl;
```

```
$ p6 dates_and_prices.pl
Cannot look up attributes in a type object
  in block <unit> at dates_and_prices.pl:17
  in any <unit-outer> at dates_and_prices.pl:1
```

**Comment by lizmat** [comment]
> On 25 May 2015, at 19:03, Andy Weidenbaum <notifications@github.com> wrote:
> 
> Basic case:
> 
> use
>  v6;
> 
> 
> subset Price of Rat where
>  * >= 0;
> 
> 
> my Price %dates_and_prices
> {Date};
> 
> my Price %dates_and_prices_from_file
> {Date};
> 
> 
> %dates_and_prices
>  =
>     Date.
> new("2014-01-01"
> ) => 13.45,
>     Date.
> new("2014-01-02"
> ) => 12.34,
>     Date.
> new("2014-01-03"
> ) => 11.11,
>     Date.
> new("2014-01-04"
> ) => 10.09,
>     Date.
> new("2014-01-05"
> ) => 9.08,
>     Date.
> new("2014-01-06"
> ) => 8.07,
>     Date.
> new("2014-01-07"
> ) => 7.06;
> 
> 
> my Price %xe{Date} = (%dates_and_prices_from_file, %dates_and_prices
> );
> 
> 
> say %xe.perl;
> $ p6 dates_and_prices.pl
> Cannot look up attributes in a type object
>   in block <unit> at dates_and_prices.pl:17
>   in any <unit-outer> at dates_and_prices.pl:1

This can be golfed down to:

my %h = my %x{Date}

and should probably be a rakudobug.



Liz

**Comment by atweiden** [comment]
Just registered this ticket at rt.perl.org: https://rt.perl.org/Ticket/Display.html?id=125249

**Comment by lizmat** [comment]
This appears to be fixed by rakudo commit 869306336fadd85320bfc1

**Comment by atweiden** [comment]
Thank you @lizmat. Switching to latest nom solves it.
